### DISCUSSION TITLE
Modern Fortran extension's dependency on the C/C++ extension(msvscode-cpptools) ...

### DISCUSSION THREAD

**Discussion by ynty** [krvajal/vscode-fortran-support#1009] [Q&A - deprecated see Fortran Discourse] [answered]
I use [clangd](https://marketplace.visualstudio.com/items?itemName=llvm-vs-code-extensions.vscode-clangd) for C/C++ coding in VSCode. But when I tries to write Fortran, I found that this extension depends on [C/C++(msvscode-cpptools)](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools-extension-pack).

 I'm confused about it, because there maybe conflicts if I install two extensions with similar features.

**Comment by gnikit** [comment, marked as answer]
The extension requires Microsoft's C/C++ extension for the debugging capabilities, we use the same debug adaptors under GCC and Clang as the C/C++ extension. I am not sure if the vscode clangd extension and Microsoft's extension cause conflicts. You will have to give it a try.

**Reply by ynty** [reply]
So, this means that Modern Fortran only uses cpptools's debugger except Microsoft's C/C++ LSP?

**Reply by gnikit** [reply]
We use only their Debug Adaptors, for LSP we use [fortls](https://fortls.fortran-lang.org/)

**Reply by ynty** [reply]
That means I can disable IntelliSense for C/C++ and use clangd when editing maybe.
Thanks a lot.

**Reply by gnikit** [reply]
I am glad this works out for you. I need to do some maintenance to the extension + add some documentation to push it over the v4.0 release finish line, but I have been unexpectedly busy this few months. I will make a note to document the reason why C/C++ is used in Modern Fortran. If it were up to me, the dependency to C/C++ would be optional, but unfortunately VS Code does not support optional dependencies.

**Reply by ynty** [reply]
Thanks.
And I tried to install both msvscode-cpptools and clangd.

clangd will show a message to disable msvscode-cpptools's IntelliSense, which will add this code in settings.json:
```json
{
    "C_Cpp.intelliSenseEngine": "disabled",
}
```
